soco/music_services/music_service.py

```diff
@@ -0,0 +1,780 @@
+                ('Username', music_service.account.username)
+            ])['SessionId']
+            session_elt = XML.Element('sessionId')
+            session_elt.text = session_id
+            credentials_header.append(session_elt)
+
+        # Anonymous auth. No need for anything further.
+        return XML.tostring(credentials_header)
+
+    def call(self, method, args=None):
+        """ Call a method on the server
+
+        Args:
+            method (str): The name of the method to call.
+             args (list): A list of (parameter, value) pairs representing
+                 the parameters of the method. Default None.
+
+        Returns:
+            (OrderedDict): An OrderedDict representing the response
+
+        Raises:
+            MusicServiceException
+
+        """
+        message = SoapMessage(
+            endpoint=self.endpoint,
+            method=method,
+            parameters=[] if args is None else args,
+            http_headers=self.http_headers,
+            soap_action="http://www.sonos.com/Services/1"
+                        ".1#{0}".format(method),
+            soap_header=self.get_soap_header(),
+            namespace=self.namespace,
+            timeout=self.timeout)
+
+        try:
+            result_elt = message.call()
+        except SoapFault as exc:
+            if 'Client.TokenRefreshRequired' in exc.faultcode:
+                log.debug('Token refresh required. Trying again')
+
+                # <detail>
+                #   <refreshAuthTokenResult>
+                #       <authToken>xxxxxxx</authToken>
+                #       <privateKey>zzzzzz</privateKey>
+                #   </refreshAuthTokenResult>
+                # </detail>
+                auth_token = exc.detail.findtext('.//authToken')
+                private_key = exc.detail.findtext('.//privateKey')
+                # We have new details - update the account
+                self.music_service.account.oa_device_id = auth_token
+                self.music_service.account.key = private_key
+                message = SoapMessage(
+                    endpoint=self.endpoint,
+                    method=method,
+                    parameters=args,
+                    http_headers=self.http_headers,
+                    soap_action="http://www.sonos.com/Services/1"
+                                ".1#{0}".format(method),
+                    soap_header=self.get_soap_header(),
+                    namespace=self.namespace,
+                    timeout=self.timeout)
+                result_elt = message.call()
+
+            else:
+                raise MusicServiceException(exc.faultstring, exc.faultcode)
+
+        # The top key in the dict will be the methodResult. Its
+        # value may be None if no results were returned.
+        result = parse(
+            XML.tostring(result_elt), process_namespaces=True,
+            namespaces={'http://www.sonos.com/Services/1.1': None}
+        ).values()[0]
```

Comment:
I'm assuming, for this to be valid, that xmltodict.parse always returns an ordereddict, although I don't think that was completely obvious from the doc string. At first, I was wondering about why you introduced the dependence here on xmltodict, if the only thing you do is to pull a single value out, but I guess that result in itself contains lots of values that we would rather have in a dict.